src/hooks/useMetamaskDetect.js

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect } from 'react'
+
+export const useMetamaskDetect = () => {
+  const [isMetaMaskInstalled, setIsMetaMaskInstalled] = useState(null)
+
+  useEffect(() => {
+    // eip6963RequestProvider timed out
+    const timeoutId = setTimeout(() => {
+      setIsMetaMaskInstalled(false)
+    }, 500)
```

Comment:
This slows down the initial page rendering, can we reduce it?

Otherwise I think everything is good, i think we can merge and improve later

Reply:
I'm going to test and see how much I can reduce it to.

Reply:
Reduced to 50ms, it's seems to be enough 👌